provider/spark.go

```diff
@@ -225,20 +235,180 @@ func NewSparkStore(storeType SparkStoreType, config SerializedConfig) (SparkStor
 	return nil, nil
 }
 
+func (s *S3Store) resourcePrefix(id ResourceID) string {
+	return fmt.Sprintf("featureform/%s/%s/%s/", id.Type, id.Name, id.Variant)
+}
+
+func (s *S3Store) ResourceKey(id ResourceID) (string, error) {
+	filePrefix := s.resourcePrefix(id)
+	fmt.Println(filePrefix)
+	objects, err := s.client.ListObjects(context.TODO(), &s3.ListObjectsInput{
```

Comment:
this would return up to 1000 records. just fyi. we will have to add support for locations with more objects.